testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -645,4 +646,112 @@ public void testReadSparkBucketedTable()
+        try {
+            onSpark().executeQuery(format("CREATE TABLE %s(" +
+                    "c_tinyint            BYTE, " +
+                    "c_smallint           SMALLINT, " +
+                    "c_int                INT, " +
+                    "c_bigint             BIGINT, " +
+                    "c_float              REAL, " +
+                    "c_double             DOUBLE, " +
+                    "c_decimal            DECIMAL(10,0), " +
+                    "c_decimal_w_params   DECIMAL(10,5), " +
+                    "c_timestamp          TIMESTAMP, " +
+                    "c_date               DATE, " +
+                    "c_string             STRING, " +
+                    "c_varchar            VARCHAR(10), " +
+                    "c_char               CHAR(10), " +
+                    "c_boolean            BOOLEAN, " +
+                    "c_binary             BINARY" +
+                    ")", tableName));
+
+            onSpark().executeQuery(format("INSERT INTO %s VALUES " +
+                    "(120, 32760, 2147483640, 9223372036854775800, 123.340, 234.560, CAST(343.0 AS DECIMAL(10, 0)), CAST(345.670 AS DECIMAL(10, 5)), TIMESTAMP '2015-05-10 12:15:30', DATE '2015-05-08', 'p1 varchar', CAST('varchar10' AS VARCHAR(10)), CAST('p1 char10' AS CHAR(10)), false, CAST('p1 binary' as BINARY))," +
+                    "(null, null, null, null, null, null, null, null, null, null, null, null, null, null, null)", tableName));
+            onSpark().executeQuery(format("ANALYZE TABLE %s COMPUTE STATISTICS FOR ALL COLUMNS", tableName));
+            assertThat(onTrino().executeQuery(format("SHOW STATS FOR %s", tableName))).containsOnly(ImmutableList.of(
+                    row("c_tinyint", null, 1.0, 0.5, null, "120", "120"),
+                    row("c_smallint", null, 1.0, 0.5, null, "32760", "32760"),
+                    row("c_int", null, 1.0, 0.5, null, "2147483640", "2147483640"),
+                    row("c_bigint", null, 1.0, 0.5, null, "9223372036854775807", "9223372036854775807"),
+                    row("c_float", null, 1.0, 0.5, null, "123.34", "123.34"),
+                    row("c_double", null, 1.0, 0.5, null, "234.56", "234.56"),
+                    row("c_decimal", null, 1.0, 0.5, null, "343.0", "343.0"),
+                    row("c_decimal_w_params", null, 1.0, 0.5, null, "345.67", "345.67"),
+                    row("c_timestamp", null, 1.0, 0.5, null, null, null),
+                    row("c_date", null, 1.0, 0.5, null, "2015-05-08", "2015-05-08"),
+                    row("c_string", 10.0, 1.0, 0.5, null, null, null),
+                    row("c_varchar", 9.0, 1.0, 0.5, null, null, null),
+                    row("c_char", 10.0, 1.0, 0.5, null, null, null),
+                    row("c_boolean", null, 1.0, 0.5, null, null, null),
+                    row("c_binary", 9.0, null, 0.5, null, null, null),
+                    row(null, null, null, null, 2.0, null, null)));
```

Comment:
data_size column shouldn't remain `null`

Reply:
`data_size` is null because of `ShowStatsRewrite` which only present row count.

Reply:
see [other](https://github.com/trinodb/trino/pull/16120/files#r1111434653)